selfservice/hook/web_hook.go

```diff
@@ -173,12 +180,63 @@ func newWebHookConfig(r json.RawMessage) (*webHookConfig, error) {
 	}, nil
 }
 
-func NewWebHook(r webHookDependencies, c json.RawMessage) *WebHook {
-	return &WebHook{r: r, c: c}
+func NewWebHook(ctx context.Context, r webHookDependencies, c json.RawMessage) *WebHook {
+	tracer := r.Tracer(ctx)
+	client := createHttpClient(tracer)
+	return &WebHook{r: r, c: c, httpClient: client}
+}
+
+// The RoundTripperFunc type is an adapter to allow the use of ordinary
+// functions as RoundTrippers. If f is a function with the appropriate
+// signature, RountTripperFunc(f) is a RoundTripper that calls f.
+type RoundTripperFunc func(req *http.Request) (*http.Response, error)
+
+// RoundTrip implements the RoundTripper interface.
+func (rt RoundTripperFunc) RoundTrip(r *http.Request) (*http.Response, error) {
+	return rt(r)
+}
+
+func createHttpClient(tracer *tracing.Tracer) *http.Client {
+	if tracer == nil || tracer.IsLoaded() == false {
+		return http.DefaultClient
+	}
+
+	delegateClient := http.DefaultTransport
+
+	var roundTripper RoundTripperFunc = func(req *http.Request) (*http.Response, error) {
```

Comment:
Could you move this to `ory/x/httpx`? It could be an option like

https://github.com/ory/x/blob/e38bb388d3f8982c170d256247a3b4f42a8ce589/httpx/resilient_client.go#L40

so for example:

```
httpx.ResilientClientWithTracer()
```

that would allow this to be used across the ecosystem :)

Reply:
I will have a look, also at the test